Resources/doc/events.rst

```diff
@@ -11,155 +11,71 @@ see Doctrine's `Event Documentation`_.
-and then `tagging`_ it with a specific tag.
+To register a service to act as an event listener you have to tag it with the
+``doctrine_mongodb.odm.event_listener`` tag.
 
-Event Listeners
----------------
-
-Use the ``doctrine_mongodb.odm.event_listener`` tag to
-register a listener. The ``event`` attribute is required and should denote
-the event on which to listen. By default, listeners will be registered with
-event managers for all connections. To restrict a listener to a single
-connection, specify its name in the tag's ``connection`` attribute.
-
-The ``priority`` attribute, which defaults to ``0`` if omitted, may be used
-to control the order in which listeners are registered. Much like Symfony's
-`event dispatcher`_, greater number will result in the listener executing
-first and listeners with the same priority will be executed in the order that
-they were registered with the event manager.
-
-Lastly, the ``lazy`` attribute, which defaults to ``false`` if omitted, may
-be used to request that the listener be lazily loaded by the event manager
-when its event is dispatched.
+Starting with Doctrine bundle 2.8, you can use the ``AsDocumentListener`` attribute to tag the service.
 
 .. configuration-block::
 
-    .. code-block:: yaml
-
-        services:
-            my_doctrine_listener:
-                class:   App\Listener\MyDoctrineListener
-                # ...
-                tags:
-                    -  { name: doctrine_mongodb.odm.event_listener, event: postPersist }
-
-    .. code-block:: xml
-
-        <service id="my_doctrine_listener" class="App\Listener\MyDoctrineListener">
-            <!-- ... -->
-            <tag name="doctrine_mongodb.odm.event_listener" event="postPersist" />
-        </service>
-
-    .. code-block:: php
+    .. code-block:: php-attributes
 
-        $definition = new Definition('App\Listener\MyDoctrineListener');
-        // ...
-        $definition->addTag('doctrine_mongodb.odm.event_listener', [
-            'event' => 'postPersist',
-        ]);
-        $container->setDefinition('my_doctrine_listener', $definition);
+        // src/App/EventListener/SearchIndexer.php
+        namespace App\EventListener;
 
-Event Subscribers
------------------
+        use Doctrine\Bundle\MongoDBBundle\Attribute\AsDocumentListener;
+        use Doctrine\ORM\Event\LifecycleEventArgs;
 
-Implement ``Doctrine\Bundle\MongoDBBundle\EventSubscriber\EventSubscriberInterface``
-and `autoconfiguration`_ to automatically register your class as a MongoODM
-event subscriber.
-
-.. code-block:: php
-
-    // src/App/EventSubscriber/MongoDB/ProductSubscriber.php
-    namespace App\EventSubscriber\MongoDB;
-
-    use App\Document\Product;
-    use Doctrine\Bundle\MongoDBBundle\EventSubscriber\EventSubscriberInterface;
-
-    class ProductSubscriber implements EventSubscriberInterface
-    {
-        public function getSubscribedEvents()
+        #[AsDocumentListener('postPersist', priority: 500, connection: 'default')]
```

Comment:
https://github.com/doctrine/DoctrineMongoDBBundle/blob/57dc5c9767ffc232e174350648b241437c02f4eb/Attribute/AsDocumentListener.php doesn't have `priority`

Reply:
Adding `priority` in https://github.com/doctrine/DoctrineMongoDBBundle/pull/822